File: DSA/pranavdixit_590014418/Day_79_Question_2.c
#include <stdio.h>
#include <stdlib.h>

struct Node {
    int data;
    struct Node *left, *right;
};

struct Node* newNode(int data) {
    struct Node* node = (struct Node*)malloc(sizeof(struct Node));
    node->data = data;
    node->left = node->right = NULL;
    return node;
}

struct Node* insert(struct Node* root, int key) {
    if (root == NULL) return newNode(key);
    if (key < root->data)
        root->left = insert(root->left, key);
    else if (key > root->data)
        root->right = insert(root->right, key);
    return root;
}

int countNodes(struct Node* root) {
    if (root == NULL) return 0;
    return 1 + countNodes(root->left) + countNodes(root->right);
}

int main() {
    struct Node* root = NULL;
    root = insert(root, 5);
    insert(root, 3);
    insert(root, 7);

    printf("Total Nodes = %d\n", countNodes(root));

    struct Node* root2 = NULL;
    root2 = insert(root2, 8);
    insert(root2, 6);
    insert(root2, 10);
    insert(root2, 4);

    printf("Total Nodes (second tree) = %d\n", countNodes(root2));

    return 0;
}

File: DSA/LakshaKundra_590013237/Day_79_Question_2.py
class Node:
    def __init__(self, val):
        self.val = val
        self.left = None
        self.right = None

def count_nodes(root):
    if not root:
        return 0
    return 1 + count_nodes(root.left) + count_nodes(root.right)

root = Node(5)
root.left = Node(3)
root.right = Node(7)
print(count_nodes(root))

root2 = Node(8)
root2.left = Node(6)
root2.right = Node(10)
root2.left.left = Node(4)
print(count_nodes(root2))
